File: JavaSRC/Item.java
import java.util.ArrayList;


public class Item {

  public String name;
  public String abbrv;

  public ArrayList<ItemInfo> equipment;


  public Item(String name, String, abbrv) {

  }

  /*
  * Passes to ItemInfo constructor
  */
  public boolean addItemInfo(String descriptor, String status, String barcode) {
    ItemInfo newItem = new ItemInfo(descriptor, status, equipment.size barcode);
    equipment.add(newItem); //TODO implement ArrayList add



  }

}

File: src/Item.js


// /**
// * Constructor for a type of item
// *
// */
// function createItem(String name, String abbrv) {
//   let item = {
//     name,
//     abbrv,
//     equipList,
//     numItems,
//   }
//
//   item.equipList = new Array();
//   item.numItems = 0;
//
//   return item;
// }
//
// /**
// * Adds a new item # available for checkout
// *
// */
// function addNewItem(String barcode, ) {
//
// }


class ItemType {



   constructor(name, abbrv) {
     this.name = name;
     this.abbrv = abbrv;
     this.equipList = new Array();
     this.numItems = 0;
   }

   function addNewItem(barcode, status) {
     numItems++;
      equipList[numItems] = new ItemInfo(barcode, status, numItems);
   }

}
